game-core/src/main/resources/META-INF/triplea/product.properties

```diff
@@ -0,0 +1 @@
+version = 1.9.0.0.@qualifier@
```

Comment:
IIntuitively I'd have used `%d` Or some other formatting placeholder instead of `@qualifier@`.
But I'm surprised to see that the code doesn't just go ahead an replaces this with _dev_ if this hasn't been replaced already.
This way there would be no need to require on a gradle build to replace this.

Reply:
> IIntuitively I'd have used `%d`

The standard `ReplaceTokens` filter uses the `@token@` format by default, and that seemed to be in fairly widespread use when I went looking around, so I didn't bother to change it. :smile:  Note also that, at least to a developer, `%d` would imply this component must be an integer, whereas, in reality, it can be an arbitrary string.  Take the `dev` qualifier, for example.

> Or some other formatting placeholder instead of `@qualifier@`.

Please see my response below to @DanVanAtta's comment.

> But I'm surprised to see that the code doesn't just go ahead an replaces this with _dev_ if this hasn't been replaced already.
> This way there would be no need to require on a gradle build to replace this.

I'm not sure the Gradle `processResources` task could be completely eliminated even if we were to add code in `ProductConfiguration#getVersion()` to replace `@qualifier@` with `dev`.  Otherwise, how would the build number from Travis ever get into _product.properties_?  Maybe I'm not fully understanding your suggestion?